File: leetcode/Medium/1667-find-kth-bit-in-nth-binary-string/1667-find-kth-bit-in-nth-binary-string.py
"""
 Problem: 1667. Find Kth Bit in Nth Binary String
 Language: python3
 Runtime: 0 ms (100.00%)
 Memory: 19.2 MB (94.49%)
 Tags: String, Recursion, Simulation
"""
class Solution:
    def findKthBit(self, n: int, k: int) -> str:
        if n == 1:
            return "0"
        
        mid = 1 << (n - 1)
        
        if k == mid:
            return "1"
        
        if k < mid:
            return self.findKthBit(n - 1, k)
        
        res = self.findKthBit(n - 1, (1 << n) - k)
        return "1" if res == "0" else "0"

File: leetcode/Medium/1667-find-kth-bit-in-nth-binary-string/1667-find-kth-bit-in-nth-binary-string.java
/*
 Problem: 1667. Find Kth Bit in Nth Binary String
 Language: java
 Runtime: 0 ms (100.00%)
 Memory: 42.5 MB (87.38%)
 Tags: String, Recursion, Simulation
*/
class Solution {
    public char findKthBit(int n, int k) {
        if (n == 1) return '0';
        
        int mid = 1 << (n - 1);
        
        if (k == mid) return '1';
        
        if (k < mid)
            return findKthBit(n - 1, k);
        
        char res = findKthBit(n - 1, (1 << n) - k);
        return res == '0' ? '1' : '0';
    }
}